streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -465,6 +467,10 @@ void handleLostAll() {
                 partitionToTask.remove(inputPartition);
             }
         }
+
+        if (processingMode == EXACTLY_ONCE_BETA) {
+            activeTaskCreator.reInitializeThreadProducer();
```

Comment:
See my other comment above: with eos-beta I think this should not be triggered by ProducerFencedException ever?